src/type_checker/mod.rs

```diff
@@ -330,6 +331,15 @@ impl<B: Backend> TypeChecker<B> {
                         }
                     };
 
+                    // store generic parameters as local vars in the fn_env
+                    for gen in &function.sig.generics {
+                        typed_fn_env.store_type(
+                            gen.to_string(),
+                            // todo: let generic vars carry their span
```

Comment:
they should carry their span by design if they are kept as identifiers

Reply:
there can be multiple spans for a same generic parameter name, such as `[[Field; N]; N]`